diff --git a/BuildNative b/BuildNative
index d6de087..a63227e 100755
--- a/BuildNative
+++ b/BuildNative
@@ -29,35 +29,7 @@ UNAME_ARCH=`uname -m`
 UNAME_OS=`uname -s | sed s/_NT-.*/_NT/`
 
 function get_arch() {
-    if [[ -n "$ARCH" ]]; then
-        echo "$ARCH"
-    elif [[ "$UNAME_OS" == "SunOS" ]]; then
-        if [ "`isainfo -n`" = "amd64" ]; then
-		    echo "x64"
-	    else
-		    echo "x86"
-	    fi
-    elif [[ "$UNAME_OS" == "Darwin" ]]; then
-	    echo "universal"
-    else
-        case $UNAME_ARCH in
-        i86pc|i386|i486|i586|i686|x86)
-	        echo x86
-	        ;;
-        x86_64|amd64)
-	        echo x64
-	        ;;
-        arm*)
-			    if [[ -d /lib/arm-linux-gnueabihf ]] ; then
-	        	echo "armhf"
-	        elif [[ -d /lib/arm-linux-gnueabi ]] ; then
-	        	echo "armel"
-    	    else
-    		    failed "Unknown Linux arm flavour: please edit $0 and Platform.java"
-	        fi
-	        ;;
-        esac
-    fi
+    echo "aarch64"
 }
 
 function get_os() {
diff --git a/pom.xml b/pom.xml
index ec9a1b4..f97fe68 100644
--- a/pom.xml
+++ b/pom.xml
@@ -87,6 +87,15 @@
 			</extension>
 		</extensions>
 		<plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.1</version>
+        <configuration>
+          <source>1.8</source>
+          <target>1.8</target>
+        </configuration>
+      </plugin>
 			<!--plugin>
 			<groupId>org.codehaus.mojo</groupId>
 			<artifactId>native-maven-plugin</artifactId>
@@ -314,6 +323,8 @@
 							<goal>jar</goal>
 						</goals>
 						<configuration>
+							<source>1.8</source>
+							<target>1.8</target>
 							<groups>
 								<group>
 									<title>Core Packages</title>
